base/linalg/generic.jl

```diff
@@ -431,16 +431,15 @@ julia> vecnorm([1 2 3 4 5 6 7 8 9])
 ```
 """
 function vecnorm(itr, p::Real=2)
-    isempty(itr) && return float(real(zero(eltype(itr))))
+    isempty(itr) && return float(norm(zero(eltype(itr))))
```

Comment:
Seems like it would be clearer to to use a return-type declaration?  Then we could just `return 0` in the empty case and `return count(...)` in the `p == 0` case.